### ISSUE TITLE
Apache Check Failure on Agent 6.16.0-1 (CentOS 7)

### ISSUE THREAD

**Issue by hybby** [DataDog/datadog-agent#4662]
The Apache Check has started to fail on Datadog Agent 6.16.0-1 on CentOS 7:

```
      Traceback (most recent call last):
        File "/opt/datadog-agent/embedded/lib/python2.7/site-packages/datadog_checks/base/checks/base.py", line 678, in run
          self.check(instance)
        File "/opt/datadog-agent/embedded/lib/python2.7/site-packages/datadog_checks/apache/apache.py", line 73, in check
          self._collect_metadata(server)
        File "/opt/datadog-agent/embedded/lib/python2.7/site-packages/datadog_checks/apache/apache.py", line 115, in _collect_metadata
          version = raw_version.split('/')[1]
      IndexError: list index out of range
```

Rolling back to 6.15.1 resolves the issue.

**Output of the info page (if this is a bug)**
```
# datadog-agent status
Getting the status from the agent.

===============
Agent (v6.16.0)
===============

  Status date: 2019-12-24 11:40:59.146491 UTC
  Agent start: 2019-12-24 11:38:57.616558 UTC
  Pid: 780
  Go Version: go1.12.9
  Python Version: 2.7.17
  Build arch: amd64
  Check Runners: 4
  Log Level: info

  Paths
  =====
    Config File: /etc/datadog-agent/datadog.yaml
    conf.d: /etc/datadog-agent/conf.d
    checks.d: /etc/datadog-agent/checks.d

  Clocks
  ======
    NTP offset: -264µs
    System UTC time: 2019-12-24 11:40:59.146491 UTC

  Host Info
  =========
    bootTime: 2019-12-24 11:38:44.000000 UTC
    kernelVersion: 3.10.0-1062.9.1.el7.x86_64
    os: linux
    platform: centos
    platformFamily: rhel
    platformVersion: 7.7.1908
    procs: 165
    uptime: 18s
    virtualizationRole: guest
    virtualizationSystem: xen

  Hostnames
  =========
    ec2-hostname: <redacted>
    hostname: <redacted>
    instance-id: <redacted>
    socket-fqdn: <redacted>
    socket-hostname: <redacted>
    host tags:
        <omitted>
    hostname provider: aws
    unused hostname providers:
      configuration/environment: hostname is empty
      gce: unable to retrieve hostname from GCE: status code 404 trying to GET http://169.254.169.254/computeMetadata/v1/instance/hostname

  Metadata
  ========
    cloud_provider: AWS
    hostname_source: aws

=========
Collector
=========

  Running Checks
  ==============

    apache (1.9.1)
    --------------
      Instance ID: apache:48188d16687255f6 [ERROR]
      Configuration Source: file:/etc/datadog-agent/conf.d/apache.d/conf.yaml
      Total Runs: 8
      Metric Samples: Last Run: 0, Total: 0
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 1, Total: 8
      Average Execution Time : 3ms
      Error: list index out of range
      Traceback (most recent call last):
        File "/opt/datadog-agent/embedded/lib/python2.7/site-packages/datadog_checks/base/checks/base.py", line 678, in run
          self.check(instance)
        File "/opt/datadog-agent/embedded/lib/python2.7/site-packages/datadog_checks/apache/apache.py", line 73, in check
          self._collect_metadata(server)
        File "/opt/datadog-agent/embedded/lib/python2.7/site-packages/datadog_checks/apache/apache.py", line 115, in _collect_metadata
          version = raw_version.split('/')[1]
      IndexError: list index out of range

    cpu
    ---
      Instance ID: cpu [OK]
      Configuration Source: file:/etc/datadog-agent/conf.d/cpu.d/conf.yaml.default
      Total Runs: 7
      Metric Samples: Last Run: 6, Total: 36
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 0s


    disk (2.5.3)
    ------------
      Instance ID: disk:e5dffb8bef24336f [OK]
      Configuration Source: file:/etc/datadog-agent/conf.d/disk.d/conf.yaml.default
      Total Runs: 8
      Metric Samples: Last Run: 60, Total: 448
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 33ms


    file_handle
    -----------
      Instance ID: file_handle [OK]
      Configuration Source: file:/etc/datadog-agent/conf.d/file_handle.d/conf.yaml.default
      Total Runs: 7
      Metric Samples: Last Run: 5, Total: 35
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 0s


    io
    --
      Instance ID: io [OK]
      Configuration Source: file:/etc/datadog-agent/conf.d/io.d/conf.yaml.default
      Total Runs: 8
      Metric Samples: Last Run: 26, Total: 190
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 0s


    load
    ----
      Instance ID: load [OK]
      Configuration Source: file:/etc/datadog-agent/conf.d/load.d/conf.yaml.default
      Total Runs: 7
      Metric Samples: Last Run: 6, Total: 42
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 0s


    memory
    ------
      Instance ID: memory [OK]
      Configuration Source: file:/etc/datadog-agent/conf.d/memory.d/conf.yaml.default
      Total Runs: 8
      Metric Samples: Last Run: 17, Total: 136
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 0s


    network (1.12.2)
    ----------------
      Instance ID: network:4568a8c8c4d5d3d5 [OK]
      Configuration Source: file:/etc/datadog-agent/conf.d/network.d/conf.yaml
      Total Runs: 8
      Metric Samples: Last Run: 32, Total: 256
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 22ms


    ntp
    ---
      Instance ID: ntp:d884b5186b651429 [OK]
      Configuration Source: file:/etc/datadog-agent/conf.d/ntp.d/conf.yaml.default
      Total Runs: 1
      Metric Samples: Last Run: 1, Total: 1
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 1, Total: 1
      Average Execution Time : 17ms


    uptime
    ------
      Instance ID: uptime [OK]
      Configuration Source: file:/etc/datadog-agent/conf.d/uptime.d/conf.yaml.default
      Total Runs: 7
      Metric Samples: Last Run: 1, Total: 7
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 0s

========
JMXFetch
========

  Initialized checks
  ==================
    no checks

  Failed checks
  =============
    no checks

=========
Forwarder
=========

  Transactions
  ============
    CheckRunsV1: 8
    Dropped: 0
    DroppedOnInput: 0
    Events: 0
    HostMetadata: 0
    IntakeV1: 2
    Metadata: 0
    Requeued: 0
    Retried: 0
    RetryQueueSize: 0
    Series: 0
    ServiceChecks: 0
    SketchSeries: 0
    Success: 18
    TimeseriesV1: 8

  API Keys status
  ===============
    API key ending with 93dfd: API Key valid

==========
Endpoints
==========
  https://app.datadoghq.com - API Key ending with:
      - 93dfd

==========
Logs Agent
==========

  Logs Agent is not running

=========
Aggregator
=========
  Checks Metric Sample: 1,288
  Dogstatsd Metric Sample: 623
  Event: 1
  Events Flushed: 1
  Number Of Flushes: 8
  Series Flushed: 1,287
  Service Check: 78
  Service Checks Flushed: 86

=========
DogStatsD
=========
  Event Packets: 0
  Event Parse Errors: 0
  Metric Packets: 622
  Metric Parse Errors: 0
  Service Check Packets: 0
  Service Check Parse Errors: 0
  Udp Bytes: 41,317
  Udp Packet Reading Errors: 0
  Udp Packets: 623
  Uds Bytes: 0
  Uds Origin Detection Errors: 0
  Uds Packet Reading Errors: 0
  Uds Packets: 0
```

**Output of Apache Check Config File**

```
# cat /etc/datadog-agent/conf.d/apache.d/conf.yaml
init_config:

instances:

  - apache_status_url: http://localhost/server-status?auto
```

Note that neither adjusting the spacing or quoting the URL resolve the issue.

**Describe what happened:**
A stack trace is thrown and Apache metrics are not recorded in `datadog-agent-6.16.0-1.x86_64` for CentOS 7.  This config file worked and these metrics were collected successfully in `datadog-agent-6.15.1-1.x86_64`.

**Describe what you expected:**
The Apache check should successfully post metrics and report OK instead of ERROR

**Steps to reproduce the issue:**
Install `datadog-agent-6.16.0-1.x86_64` and configure `/etc/datadog-agent/conf.d/apache.d/conf.yaml` as so:

```
init_config:

instances:

  - apache_status_url: http://localhost/server-status?auto
```

Note that neither adjusting the spacing or quoting the URL resolve the issue.

**Additional environment details (Operating System, Cloud provider, etc):**
AWS, CentOS 7

```
[root@ip-10-27-102-241 ~]# cat /etc/redhat-release
CentOS Linux release 7.7.1908 (Core)

[root@ip-10-27-102-241 ~]# uname -a
Linux ip-10-27-102-241 3.10.0-1062.9.1.el7.x86_64 #1 SMP Fri Dec 6 15:49:49 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

Apache version:

```
[root@ip-10-27-91-194 ~]# rpm -qa | grep httpd-
httpd-tools-2.4.6-90.el7.centos.x86_64
httpd-2.4.6-90.el7.centos.x86_64

[root@ip-10-27-91-194 ~]# httpd -V
Server version: Apache/2.4.6 (CentOS)
Server built:   Aug  8 2019 11:41:18
Server's Module Magic Number: 20120211:24
Server loaded:  APR 1.4.8, APR-UTIL 1.5.2
Compiled using: APR 1.4.8, APR-UTIL 1.5.2
Architecture:   64-bit
Server MPM:     event
  threaded:     yes (fixed thread count)
    forked:     yes (variable process count)
Server compiled with....
 -D APR_HAS_SENDFILE
 -D APR_HAS_MMAP
 -D APR_HAVE_IPV6 (IPv4-mapped addresses enabled)
 -D APR_USE_SYSVSEM_SERIALIZE
 -D APR_USE_PTHREAD_SERIALIZE
 -D SINGLE_LISTEN_UNSERIALIZED_ACCEPT
 -D APR_HAS_OTHER_CHILD
 -D AP_HAVE_RELIABLE_PIPED_LOGS
 -D DYNAMIC_MODULE_LIMIT=256
 -D HTTPD_ROOT="/etc/httpd"
 -D SUEXEC_BIN="/usr/sbin/suexec"
 -D DEFAULT_PIDLOG="/run/httpd/httpd.pid"
 -D DEFAULT_SCOREBOARD="logs/apache_runtime_status"
 -D DEFAULT_ERRORLOG="logs/error_log"
 -D AP_TYPES_CONFIG_FILE="conf/mime.types"
 -D SERVER_CONFIG_FILE="conf/httpd.conf"
```

**Comment by hybby** [comment]
I've further debugged to show that the contents of the `server` variable is simply `Apache`:

I edited from line 72 of `/opt/datadog-agent/embedded/lib/python2.7/site-packages/datadog_checks/apache/apache.py` as so:

```
        if server:
            self.log.debug("Server: {}".format(server))
            self._collect_metadata(server)
```

Debug output shows:

```
2019-12-24 12:20:08 UTC | CORE | DEBUG | (pkg/collector/python/datadog_agent.go:122 in LogMessage) | apache:48188d16687255f6 | (apache.py:73) | Server: Apache
```

This seems to be consistent with the content of the headers of the request being made:

```
# curl -I http://localhost/server-status?auto
HTTP/1.1 200 OK
Date: Tue, 24 Dec 2019 12:22:26 GMT
Server: Apache
Content-Length: 2253
Content-Type: text/plain; charset=ISO-8859-1
```

It seems like the `_collect_metadata` function is expecting Server to be a string that contains the Apache version which doesn't seem to be the case on my system.

**Comment by hybby** [comment]
Looks like it may be fixed as part of: https://github.com/DataDog/integrations-core/pull/5252

**Comment by AlexandreYang** [comment]
@hybby Yes, this is fixed by the PR you mentioned.

It will be included in Agent 7.16.1 (release very soon).

You can also install the newer version of the integration without upgrading the agent itself by using:
```
datadog-agent integration install datadog-apache==1.9.2
```

The way the `datadog-agent integration` is executed might be different depending on your system: https://docs.datadoghq.com/agent/guide/agent-commands/?tab=agentv6v7#agent-information
